src/builder/create_attachment.rs

```diff
@@ -114,6 +144,45 @@ impl<'a> CreateAttachment<'a> {
     }
 }
 
+#[derive(Clone, Debug, Serialize)]
+#[serde(transparent)]
+pub struct ImageData(String);
+
+impl ImageData {
+    /// Constructs image data from a base64-encoded blob of data. The string must be a valid data
+    /// URI, and must be encoded with base64, for example:
+    ///
+    /// ```
+    /// use serenity::builder::ImageData;
+    ///
+    /// let s = "data:image/png;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs=";
+    /// assert!(ImageData::from_base64(s).is_ok());
+    /// ```
+    ///
+    /// # Errors
+    ///
+    /// Returns a [`Error::Url`] if the string is not a valid data URI. See the [Discord
+    /// docs](https://discord.com/developers/docs/reference#image-data).
+    pub fn from_base64(s: &str) -> Result<Self> {
+        let url = Url::parse(s).map_err(|_| UrlError::InvalidDataURI)?;
+
+        let err = Error::Url(UrlError::InvalidDataURI);
+        if url.scheme() != "data" {
+            return Err(err);
+        }
+
+        let Some((mimetype, encoding)) = url.path().split_once(';') else {
+            return Err(err);
+        };
```

Comment:
Since this is already doing manual parsing for the data URI, can we drop the usage of `Url`? It's quite an expensive type and requires reallocating the buffer which in past has been a performance problem right?